### PR TITLE
[CAMEL-21039] Align quarkus generated k8s resources with other runtimes

### PR DESCRIPTION
**Pull request by tdiesler**
https://issues.apache.org/jira/browse/CAMEL-21039

### CODE REVIEW

**Comment by tdiesler** [comment]
Currently, we make a copy of the respective *-pom.tmpl for each runtime. i.e. you will find *-kubernetes-pom.tmpl for each runtime. Then, it is up to JKube to package the executable jar for use in the Docker container, build the image and finally generate the k8s manifest.

Hence, when you do camel kubernetes export, you'll get an executable jar, which you can run as documented [here](https://github.com/tdiesler/camel-cloud-examples/tree/main/main/timer-log). It is however not the same as you would get from plain camel export.

The idea was to implement camel kubernetes export, with as little as possible changes to the plain export/run.

PS: Indeed, the package as runner jar step was delete by mistake.

File: dsl/camel-jbang/camel-jbang-core/src/main/resources/templates/main-kubernetes-pom.tmpl [hunk 1]
```diff
@@ -105,7 +105,7 @@
                     </descriptors>
                     <archive>
                         <manifest>
-                            <mainClass>{{ .MainClassname }}</mainClass>
+                            <mainClass>examples.timerlog.CamelApplication</mainClass>
```
Comment (davsclaus) [inline dsl/camel-jbang/camel-jbang-core/src/main/resources/templates/main-kubernetes-pom.tmpl:108]: This seems like a mistake

**Review by oscerd — APPROVED** [review]

**Review by davsclaus — APPROVED** [review]

**Comment by davsclaus** [comment]
@christophd do you have any feedback on this PR ?

**Review by gansheer — APPROVED** [review]

**Review by davsclaus — APPROVED** [review]

**Comment by tdiesler** [comment]
Yes, otherwise it would be in draft or had unresolved comments

File: dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/KubernetesExport.java [hunk 2]
```diff
@@ -260,51 +260,30 @@ public Integer export() throws Exception {
...
             // Quarkus specific dependencies
+            // [TODO] migrate to JKube
             if (clusterType != null && clusterType.equals("openshift")) {
                 addDependencies("io.quarkus:quarkus-openshift");
                 if (imageBuilder == null) {
                     // use s2i image builder as a default on OpenShift
                     imageBuilder = "s2i";
                 }
-            } else {
-                addDependencies("io.quarkus:quarkus-kubernetes");
```
Comment (christophd) [inline dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/KubernetesExport.java:280]: why did you remove this?

File: dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/KubernetesExport.java [hunk 3]
```diff
@@ -260,51 +260,30 @@ public Integer export() throws Exception {
...
-                imagePullPolicy = StringHelper.camelCaseToDash(imagePullPolicy);
-            }
             buildProperties.add("%s.kubernetes.image-pull-policy=%s".formatted(propPrefix, imagePullPolicy));
         }
 
         // Quarkus Runtime specific
         if (runtime == RuntimeType.quarkus) {
 
             // Quarkus specific dependencies
+            // [TODO] migrate to JKube
```
Comment (christophd) [inline dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/KubernetesExport.java:270]: Is this still a TODO?

**Review by christophd — APPROVED** [review]
Just some clarification questions

**Comment by christophd** [comment]
@tdiesler sorry, I am late to the review party. I only have two questions to be sure that I understand correctly. many thanks

File: dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/KubernetesExport.java (hunk 3 above)
Comment (tdiesler) [reply dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/KubernetesExport.java:270]: It is part of https://issues.apache.org/jira/browse/CAMEL-21013 which I'm also working on

File: dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/KubernetesExport.java (hunk 2 above)
Comment (tdiesler) [reply dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/KubernetesExport.java:280]: because, as we discussed, this is now uniformly handled by jkube

Comment (davsclaus) [reply dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/KubernetesExport.java:280]: Hmm we should surely allow users to use the _quarkus standard_ for running on k8s. And would that not be that you should use their quarkus-kubernetes extension? So you can do all their configurations as well according to their guides.
https://quarkus.io/guides/deploying-to-kubernetes

**Review by davsclaus — CHANGES_REQUESTED** [review]
need to be cliarified more

**Comment by davsclaus** [comment]
ups did my review change cause the comments from christopher and me dissapear ?

File: dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/KubernetesExport.java (hunk 2 above)
Comment (christophd) [reply dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/KubernetesExport.java:280]: @tdiesler it confuses me that the `io.quarkus:quarkus-openshift` is being added then when cluster-type is "openshift". how is that going to work when we remove all the quarkus settings?

Comment (christophd) [reply dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/KubernetesExport.java:280]: @davsclaus the downside of this is that the export for runtime `quarkus` would be different compared to the other runtimes `main` and `spring-boot`. the idea of this PR is to find a common ground for all runtimes so that all exports use pure jKube plugin functionalities. but yes this also means users cannot use the quarkus Kubernetes extension configuration settings OOTB after export @tdiesler I got that right?

File: dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/KubernetesExport.java (hunk 3 above)
Comment (christophd) [reply dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/KubernetesExport.java:270]: should we the wait for CAMEL-21013 to be resolved first? Otherwise this PR might break the deployment to OpenShift right?

File: dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/KubernetesExport.java (hunk 2 above)
Comment (tdiesler) [reply dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/KubernetesExport.java:280]: Yes, the idea was that independent of runtime there is just one way on configuring k8s/openshift resources. The Openshift related code that is currently marked as TODO is untested legacy (which could also be removed in this PR). Support for Openshift will come back with [CAMEL-21013](https://issues.apache.org/jira/browse/CAMEL-21013), also uniformly through JKube.

**Comment by davsclaus** [comment]
Okay we can all these runtimes. But later we should really use the _powers of the runtimes_ and let all this k8s configuration be done how these runtimes do it - Q and SB may not be similar, but our users want to use the standard way, instead of some custom way.

And the export is just a way to move on from prototype to a real project where the developer can then be on a good start to finish the project

File: dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/KubernetesExport.java (hunk 3 above)
Comment (tdiesler) [reply dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/KubernetesExport.java]: Sure, we could wait until openshift is done as well. If we're not going to release any time soon, I'd prefer to do this iteratively and leave openshift out for now. I can continue to look at it next week - today I couldn't get cluster through the chat bot.

**Comment by tdiesler** [comment]
Needs to get revisited. We want to stick with Quarkus for image building and k8s resource generation and use JKube for main and spring-boot
